README.md

```diff
@@ -3,159 +3,229 @@ Zeebe Test Container
 use containers for your tests.
 
-Features
-========
-
-- [x] Start a Zeebe broker container with configurable environment
-- [x] Start a Zeebe gateway container with configurable environment
-
-Planned
-=======
-
-Current plans are to add more tests and QoL features:
-
-- [ ] Gateway tests
-- [ ] Verification API
-- [ ] Client builder
-- [ ] Cluster rule
-
 Supported Zeebe versions
 ========================
 
-> Supported versions are those we actively test against; in general the library should be
-compatible with most Zeebe versions `>= 0.20.x`.
+> **NOTE**: version 1.0 is incompatible with Zeebe versions pre 0.22.x
 
-- 0.20.1
-- 0.21.1
 - 0.22.1
+- 0.23.3
+- 0.24.0
 
-Quickstart
-==========
+# Quickstart
 
 Add the project to your dependencies:
 
 ```xml
 <dependency>
   <groupId>io.zeebe</groupId>
   <artifactId>zeebe-test-container</artifactId>
-  <version>0.31.1</version>
+  <version>1.0.0</version>
 </dependency>
 ```
 
-You can then use `ZeebeBrokerContainer` and `ZeebeGatewayContainer` as you would any `GenericContainer`.
-
-Example usage
-=============
+## junit4
 
-Import `zeebe-test-container` to your project, and you can use the containers in your
-tests as:
+If you're using junit4, you can add the container as a rule: it will be started and closed around
+each test execution. You can read more about Testcontainers and junit4 [here](https://www.testcontainers.org/test_framework_integration/junit_4/).
 
-### Simple broker with embedded gateway
 ```java
-class MyFeatureTest {
+package com.acme.zeebe;
+
+import io.zeebe.client.ZeebeClient;
+import io.zeebe.containers.ZeebeContainer;
+import org.junit.Test;
+import org.junit.Rule;
+public class MyFeatureTest {
   @Rule
-  public ZeebeBrokerContainer zeebe = new ZeebeBrokerContainer();
+  public ZeebeContainer zeebeContainer = new ZeebeContainer();
 
   @Test
-  public void shouldTestMyFeature() {
-    // create a client to connect to the gateway
-    final ZeebeClient client =
-        ZeebeClient.newClientBuilder()
-            .brokerContactPoint(broker.getExternalAddress(ZeebePort.GATEWAY))
-            .build();
-
-    // test stuff
-    // ...
-  }
-}
-```
+  public void shouldConnectToZeebe() {
+    // given
+    final ZeebeClient client = ZeebeClient.newClientBuilder()
+      .brokerContactPoint(zeebeContainer.getExternalGatewayAddress())
+      .usePlaintext();
 
-### Configured broker with embedded gateway
-```java
-class MyFeatureTest {
-  @Rule
-  public ZeebeBrokerContainer zeebe = new ZeebeBrokerContainer()
-    .withPartitionCount(3)
-    .withReplicationFactor(1);
+    // when
+    // do something
 
-  @Test
-  public void shouldTestMyFeature() {
-    // create a client to connect to the gateway
-    final ZeebeClient client =
-        ZeebeClient.newClientBuilder()
-            .brokerContactPoint(broker.getExternalAddress(ZeebePort.GATEWAY))
-            .build();
-
-    // test stuff
-    // ...
+    // then
+    // verify
```

Comment:
Hm, the verification is highly dependent on your test, no? In our own tests we'd probably peek into the exported stream, but others might do wildly different verifications. That's not to say there isn't something we could write down here, just that I wasn't sure what to write and figured it could be left to the reader

Reply:
I agree. I thought to give an example of deploying a workflow and creating instances, but this takes quite some code to achieve and might take away from the example. What do you think?

```
// given
final var client = ZeebeClient.newClientBuilder()
  .brokerContactPoint(zeebeContainer.getExternalGatewayAddress())
  .usePlaintext();

// when
// do something (e.g. deploy a workflow)
final var deploymentEvent = client.newDeployCommand()
    .addWorkflowModel(process, "process.bpmn")
    .send()
    .join();

// then
// verify (e.g. we can create an instance of the deployed workflow)
final var workflowInstanceResult = client.newCreateInstanceCommand()
    .bpmnProcessId("process")
    .latestVersion()
    .withResult()
    .send()
    .join();
Assertions.assertThat(workflowInstanceResult.getWorkflowKey())
  .isEqualTo(deploymentEvent.getWorkflows().get(0).getWorkflowKey);
```

Reply:
Hm, that's simple enough, I like it :+1: